Fix WebUI API doesn't update IP and Port

myfingerhurt opened the pull request:
Add a checker which prevents of adding your own gpg_id.
Add an inactive friends list based on last seen Unix timestamp.
Add a method of removing a specific peer based on peer_id.

csoler commented:
what do you mean by "not adding your own PGP Id" ? You should be able to add yourself as a friend when you have multiple locations.

myfingerhurt commented:
As fas as I know, the same pgp_id will discorvery themself and add online subnodes automaticlly even you remove one of them their just keep coming back, but if you accidently add your own peer_id on the WebUI, the peer list will show two identical PGP names and its subnodes, untill the retroshare gets restarted.  

And this modification don't add the same peer_id as long as it belongs to your current node, it doesn't mean you can't add your new nodes with the same pgp_id but different peer_id.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+ *  
+ *  \details 
+ */
+void PeersHandler::handleRemoveNode(Request &req, Response &resp)
```
Don't like this approach, the function should work with location or with PGP id only, having so different behaviour depending on how parameters are passed is confusing, more over there is already another way to remove a PGP friend via API and it's 

/peers/${PGP_ID}/delete

So in my opinion this function should be for location only

myfingerhurt replied:
I'm trying to replace the `/peers/${PGP_ID}/delete` eventually.
And both removing `pgp_id` and removing `peer_id` are so similar, I think of putting them together may be more simplified.

Well, I can split them into two in the next commit.

G10h4ck replied:
Please do that ;)

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+ *  \details Combine this with handleRemoveNode, 
+ *  batch remove inactive users are possible and convenient.
+ */
+void PeersHandler::handleGetInactiveUsers(Request &req, Response &resp)
```
As the API is used as a gateway to other languages too that may have different way of representing the date the representation format should be specified in the documentation

myfingerhurt replied:
Add Unix Time notes.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -624,34 +632,127 @@ void PeersHandler::handleWildcard(Request &req, Response &resp)
+			//std::string debug_string = " ";
+			RsPgpId own_pgp = mRsPeers->getGPGOwnId();
+			RsPeerId ownpeer_id = mRsPeers->getOwnId();
+			do{
```
please use Allman indent style as per http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -624,34 +632,127 @@ void PeersHandler::handleWildcard(Request &req, Response &resp)
+					
+					std::cerr << "handleWildcard addPeer: is own pgp_id, update only" << std::endl;
+					
+				}else{
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -624,34 +632,127 @@ void PeersHandler::handleWildcard(Request &req, Response &resp)
+				
+				std::cerr << "handleWildcard addPeer: peerDetailsToStream" << std::endl;
+				
+				if (!peerDetails.location.empty()) {
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -624,34 +632,127 @@ void PeersHandler::handleWildcard(Request &req, Response &resp)
+						std::cerr << " Port: " << peerDetails.localPort << std::endl;
+						mRsPeers->setLocalAddress(peerDetails.id, peerDetails.localAddr, peerDetails.localPort);
+					}
+					if (!peerDetails.dyndns.empty()) {
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+	RsPeerId peerId(ssl_peer_id);
+	RsPgpId gpgId(gpg_id);
+	
+	if(!gpgId.isNull()){
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+		mRsPeers->removeFriend(gpgId);
+		resp.mDataStream << makeKeyValue("gpg_id", gpg_id);
+		resp.setOk(); 
+	}else if(!peerId.isNull()){
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+		mRsPeers->removeFriendLocation(peerId);
+		resp.mDataStream << makeKeyValue("peer_id", ssl_peer_id);
+		resp.setOk();        
+	}else{
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+	
+	uint32_t before = strtoul(datetime.c_str(), NULL, 10);
+	
+	if(datetime.empty()){
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+		RsPeerDetails details;
+		mRsPeers->getPeerDetails(*lt, details);
+		// Search for exist pgp_id, update the last connect time
+		if(mapRsPgpId.find(details.gpg_id) != mapRsPgpId.end()){
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck commented on libresapi/src/api/PeersHandler.cpp:
```diff
@@ -691,6 +792,107 @@ void PeersHandler::handleExamineCert(Request &req, Response &resp)
+		if(mapRsPgpId.find(details.gpg_id) != mapRsPgpId.end()){
+			if(mapRsPgpId[details.gpg_id] < details.lastConnect)
+				mapRsPgpId[details.gpg_id] = details.lastConnect;
+		}else{
```
Please use Allman indent style http://retroshare.readthedocs.io/en/latest/developer/coding/#coding-style

myfingerhurt replied:
Done.

G10h4ck reviewed (changes requested):
Nice work, please do some changes as per comments so we can integrate it into the code easly

G10h4ck commented:
Plese just drop the unwanted commits, not revert them to avoid history bloating as this isn't in master branch yet history rewriting is not a problem

just do
```
git reset --hard 6cb4d36e00ff49e5a370dd3a60dfcf62e8fed015
git push -f
```

Commit:
Add a checker which prevents of adding your own gpg_id.
Add an inactive friends list based on last seen Unix timestamp.
Add a method of removing a specific peer based on peer_id.

As fas as I know, the same pgp_id will discorvery themself and add online subnodes automaticlly even you remove one of them their just keep coming back, but if you accidently add your own peer_id on the WebUI, the peer list will show two identical PGP names and its subnodes, untill the retroshare gets restarted.

And this modification don't add the same peer_id as long as it belongs to your current node, it doesn't mean you can't add your new nodes with the same pgp_id but different peer_id.

Honor the Allman indent style

myfingerhurt commented:
I did what you had suggested.  Does it look better or worse?
The previous code seems not following the Allman indent style. I didn't touch that, but the copy and paste in notepad++ unified the indent style automatically.
```
git reset --hard xxxxxxxxxxxxxxxxx
git push -f
```

G10h4ck commented:
Ok now please fix the Allman style thing ;)

myfingerhurt commented:
Well, I believe I have fixed Allman style things with my last commit.
Is there anything I haven't done? I'm trying to make git to ignore leading spaces and tabs or generate a patch file without indentation, it seems none exists.

But with `?w=1`  it is better looking.

https://github.com/RetroShare/RetroShare/pull/1014/commits/f3485580773eaed3086d96219a11d21a3535a428?w=1

G10h4ck commented:
The indentation fixes commit is not included in the PR probably it got caught in the revert/reset crossfire, you can cherry pick it probably and push again (no need for -f this time)

have you splitted PeersHandler::handleRemoveNode as requested?

G10h4ck commented:
I have done the required modifications and merged it afterwards, thanks you for the contribution, please in the future use the qtcreator coding style file provided in the documentation so it is easier to merge your code ;)
